gr-blocks/lib/udp_source_impl.h

```diff
@@ -38,27 +41,28 @@ namespace gr {
       int     d_payload_size; // maximum transmission unit (packet length)
       bool    d_eof;          // look for an EOF signal
       bool    d_connected;    // are we connected?
-      char   *d_rxbuf;        // get UDP buffer items
-      char   *d_residbuf;     // hold buffer between calls
-      ssize_t d_residual;     // hold information about number of bytes stored in residbuf
-      ssize_t d_sent;         // track how much of d_residbuf we've outputted
+      std::vector<std::unique_ptr<char[]>> d_rxbufs;
+      std::vector<std::unique_ptr<char[]>> d_residbufs;
+      std::vector<ssize_t> d_residuals;
+      std::vector<ssize_t> d_sents;
 
       static const int BUF_SIZE_PAYLOADS; //!< The d_residbuf size in multiples of d_payload_size
 
       std::string d_host;
       unsigned short d_port;
 
-      boost::asio::ip::udp::socket *d_socket;
-      boost::asio::ip::udp::endpoint d_endpoint;
-      boost::asio::ip::udp::endpoint d_endpoint_rcvd;
+      std::vector<std::unique_ptr<boost::asio::ip::udp::socket> > d_sockets;
```

Comment:
Just a minor nit, but `> >` is no more necessary since C++11.

Reply:
Oh, I'll remove it. I'm just used to pre-C++11 (cautions) coding.